src/backend/distributed/connection/connection_management.c

```diff
@@ -764,3 +772,54 @@ AfterXactHostConnectionHandling(ConnectionHashEntry *entry, bool isCommit)
 		}
 	}
 }
+
+
+/*
+ * SetCitusNoticeProcessor sets the NoticeProcessor to DefaultCitusNoticeProcessor
+ */
+void
+SetCitusNoticeProcessor(MultiConnection *connection)
+{
+	StringInfo nodeAddress = makeStringInfo();
+	char *hostName = connection->hostname;
+	uint32 nodePort = connection->port;
+
+	appendStringInfo(nodeAddress, "%s:%d", hostName, nodePort);
+
+	PQsetNoticeProcessor(connection->pgConn, DefaultCitusNoticeProcessor,
+						 MemoryContextStrdup(ConnectionContext, nodeAddress->data));
```

Comment:
Allocating it in `ConnectionContext` means this will not be freed until the end of the session. Maybe just use the `MultiConnection` as the context pointer instead of a separate string?